modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceUtil.java

```diff
@@ -270,7 +271,9 @@ public static CompletableFuture<?>[] triggerAllTablePartitionsRebalance(
             int finalPartId = partId;
 
             futures[partId] = tableAssignmentsFut.thenCompose(tableAssignments ->
-                    updatePendingAssignmentsKeys(
+                    // TODO https://issues.apache.org/jira/browse/IGNITE-19763 We should distinguish empty stable assignments on
+                    // TODO node recovery in case of interrupted table creation, and moving from empty assignments to non-empty.
+                    tableAssignments.isEmpty() ? nullCompletedFuture() : updatePendingAssignmentsKeys(
```

Comment:
What  `tableAssignments.isEmpty()` will return if no one has previously wrote the stable assignments?

Reply:
1. It will return true
2. Its happening on recovery, if there is a table that is not fully created yet, and there are no stable assignments written at this moment.

Reply:
> 2. Its happening on recovery, if there is a table that is not fully created yet, and there are no stable assignments written at this moment.


What will guarantee that we will not skip pending keys in that case? I mean the situation when on node recovery we have both empty assignmetns.stable and trigger for assignments.pending calculation.

Reply:
Discussed in private.